core/src/main/java/org/apache/gravitino/cache/EntityCacheKey.java

```diff
@@ -30,6 +30,24 @@ public class EntityCacheKey {
   private final NameIdentifier identifier;
   private final Entity.EntityType type;
   private final SupportsRelationOperations.Type relationType;
+  private final Boolean allFields;
+
+  /**
+   * Creates a new instance of {@link EntityCacheKey} with the given arguments.
+   *
+   * @param ident The identifier of the entity.
+   * @param type The type of the entity.
+   * @param relationType The type of the relation, it can be null.
+   * @param allFields Whether to retrieve all fields of the entity.
+   * @return A new instance of {@link EntityCacheKey}.
+   */
+  public static EntityCacheKey of(
+      NameIdentifier ident,
+      Entity.EntityType type,
+      SupportsRelationOperations.Type relationType,
+      boolean allFields) {
+    return new EntityCacheKey(ident, type, relationType, allFields);
```

Comment:
If `allFields` is false, then the entities retrieved are the same. If I call this API several times? How many fields are fetched from storage if `allFields` is false.  Are these fields constant?

Reply:
Hi @yuqi1129 ,

I understand your concern now. Sorry for misunderstanding it as a three-state `true`, `false`, and `none` state.

I plan to remove the cache of `allFields == false` and only cache the value when `allFields` is `true`.
Could this logic match our design? 😊

Best Regards,
Edward

Reply:
why not cache when `allFields==true`?